x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -0,0 +1,580 @@
+  | w/o historical versions  | new    | total: N, succeeded: N   |
+```
+
+Notes:
+
+- Legacy API:
+  - install: `PUT /api/detection_engine/rules/prepackaged`
+- New API:
+  - install: `POST /internal/detection_engine/prebuilt_rules/installation/_perform`
+
+#### **Scenario: API can install prebuilt rules that are not yet installed**
+
+**Coverage**: 4 integration tests with mock rules.
+
+```Gherkin
+Given the package <package_type> is installed
+And the package contains N rules
+When user installs all rules via install <api>
+And deletes one of the installed rules
+And gets prebuilt rules status via status <api>
+Then the endpoint should return 200 with <status_response>
+When user installs all rules via install <api> again
+Then the endpoint should return 200 with <install_response>
+
+Examples:
+  | package_type             | api    | status_response  | install_response         |
+  | with historical versions | legacy | not_installed: 1 | installed: 1, updated: 0 |
+  | w/o historical versions  | legacy | not_installed: 1 | installed: 1, updated: 0 |
+  | with historical versions | new    | to_install: 1    | total: 1, succeeded: 1   |
+  | w/o historical versions  | new    | to_install: 1    | total: 1, succeeded: 1   |
+```
+
+Notes:
+
+- Legacy API:
+  - install: `PUT /api/detection_engine/rules/prepackaged`
+  - status: `GET /api/detection_engine/rules/prepackaged/_status`
+- New API:
+  - install: `POST /internal/detection_engine/prebuilt_rules/installation/_perform`
+  - status: `GET /internal/detection_engine/prebuilt_rules/status`
+
+#### **Scenario: API can upgrade prebuilt rules that are outdated**
+
+**Coverage**: 4 integration tests with mock rules.
+
+```Gherkin
+Given the package <package_type> is installed
+And the package contains N rules
+When user installs all rules via install <api>
+And new X+1 version of a rule asset <assets_update>
+And user gets prebuilt rules status via status <api>
+Then the endpoint should return 200 with <status_response>
+When user upgrades all rules via upgrade <api>
+Then the endpoint should return 200 with <upgrade_response>
+
+Examples:
+  | package_type             | api    | assets_update | status_response | upgrade_response         |
+  | with historical versions | legacy | gets added    | not_updated: 1  | installed: 0, updated: 1 |
+  | w/o historical versions  | legacy | replaces X    | not_updated: 1  | installed: 0, updated: 1 |
+  | with historical versions | new    | gets added    | to_upgrade: 1   | total: 1, succeeded: 1   |
+  | w/o historical versions  | new    | replaces X    | to_upgrade: 1   | total: 1, succeeded: 1   |
+```
+
+TODO: Check why for the legacy API Dmitrii has added 2 integration tests for `rule package with historical versions` instead of 1:
+
+- `should update outdated prebuilt rules when previous historical versions available`
+- `should update outdated prebuilt rules when previous historical versions unavailable`
```

Comment:
TODO: @banderror sync with @jpdjere

Reply:
Synced. The TODO is left for @jpdjere to look into this when working on adding more tests in a follow-up PR.